src/Graphics/Vty/Input/Classify.hs

```diff
@@ -45,40 +48,40 @@ compile table = cl' where
                 -- H: There will always be one match. The prefixSet
                 -- contains, by definition, all prefixes of an event.
                 False ->
-                    let inputPrefixes = reverse $ take maxValidInputLength $ tail $ inits inputBlock
+                    let inputPrefixes = reverse $ take maxValidInputLength $ tail $ BS8.inits inputBlock
                     in case mapMaybe (\s -> (,) s `fmap` M.lookup s eventForInput) inputPrefixes of
-                        (s,e) : _ -> Valid e (drop (length s) inputBlock)
+                        (s,e) : _ -> Valid e (BS8.drop (BS8.length s) inputBlock)
                         -- neither a prefix or a full event.
                         [] -> Invalid
 
-classify :: ClassifyMap -> String -> KClass
+classify :: ClassifyMap -> ByteString -> KClass
 classify table =
     let standardClassifier = compile table
-    in \s -> case s of
+    in \s -> case BS.uncons s of
         _ | bracketedPasteStarted s ->
             if bracketedPasteFinished s
             then parseBracketedPaste s
             else Prefix
-        _ | isMouseEvent s   -> classifyMouseEvent s
-        _ | isFocusEvent s   -> classifyFocusEvent s
-        c:cs | ord c >= 0xC2 -> classifyUtf8 c cs
-        _                    -> standardClassifier s
+        _ | isMouseEvent s      -> classifyMouseEvent s
+        _ | isFocusEvent s      -> classifyFocusEvent s
+        Just (c,cs) | c >= 0xC2 -> classifyUtf8 c cs
+        _                       -> standardClassifier s
 
-classifyUtf8 :: Char -> String -> KClass
+classifyUtf8 :: Word8 -> ByteString -> KClass
 classifyUtf8 c cs =
-  let n = utf8Length (ord c)
-      (codepoint,rest) = splitAt n (c:cs)
+  let n = utf8Length c
+      (codepoint,rest) = BS8.splitAt (n - 1) cs
 
       codepoint8 :: [Word8]
-      codepoint8 = map (fromIntegral . ord) codepoint
+      codepoint8 = c:BS.unpack codepoint
 
   in case decode codepoint8 of
-       _ | n < length codepoint -> Prefix
-       Just (unicodeChar, _)    -> Valid (EvKey (KChar unicodeChar) []) rest
+       _ | n < BS.length codepoint + 1 -> Prefix
```

Comment:
@jtdaugherty This seems wrong, but the original code seems wrong too? Length of codepoint is never going to be greater than n. I guess it was supposed to be `>`?

Reply:
Frankly I have no idea. That code almost certainly predates my maintenance of the library..

Reply:
I'll leave the logic as is for now.